### ISSUE TITLE
package failed : failed to create directory ...... and resolved 

### ISSUE THREAD

**Issue by UEBoy2019** [secretlocation/azure-kinect-unreal#15]
### Bug

when i package to windows platform and  will failed 
### Logs
failed to create directory:     \azure-kinect-unreal-master\Saved\StagedBuilds\WindowsNoEditor\AzureKinect\Binaries\Win64

```

### SDK and UE4 versions

| UE4 | Azure Kinect | AK Body Tracking |
|:---:|:---:|:---:|
| 4.25.3 |  1.4.1 | 1.0.1 |

**the way of resolution:**

azure-kinect-unreal-master\Plugins\AzureKinectUnreal\Source\AzureKinectUnreal\AzureKinectUnreal.Build.cs
// Ensure that the DLLs are staged along with the executable
//RuntimeDependencies.Add("$(PluginDir)/Binaries/Win64/k4a.dll");
//RuntimeDependencies.Add("$(PluginDir)/Binaries/Win64/k4abt.dll");
//RuntimeDependencies.Add("$(PluginDir)/Binaries/Win64/depthengine_2_0.dll");
//RuntimeDependencies.Add("$(PluginDir) / Binaries / Win64 /opencv_world430.dll");

to exchange

RuntimeDependencies.Add(Path.Combine(AzureKinectBinaryPath, "k4a.dll"));
RuntimeDependencies.Add(Path.Combine(AzureKinectBinaryPath, "k4abt.dll"));
RuntimeDependencies.Add(Path.Combine(AzureKinectBinaryPath, "depthengine_2_0.dll"));
RuntimeDependencies.Add(Path.Combine(AzureKinectBinaryPath, "opencv_world430.dll"));

**Comment by qimenlang** [comment]
it works ,after copping k4a.dll\k4abt.dll\depthengine_2_0.dll to package_folder\WindowsNoEditor\AzureKinect\Binaries\Win64,  cool!

**Comment by jdamdami** [comment]
Hi,

I have similar issue here : 

UATHelper: Packaging (Windows (64-bit)): ERROR: Failed to copy D:\UPROJECTS\CHANEL_TryOn_08092022\Plugins\AzureKinectUnreal\Binaries\Win64\k4a.dll to D:\UPROJECTS\CHANEL_TryOn_08092022\Saved\StagedBuilds\WindowsNoEditor\Chanel_TryOn\Plugins\AzureKinectUnreal\Binaries\Win64\k4a.dll
UATHelper: Packaging (Windows (64-bit)): ERROR: Failed to copy D:\UPROJECTS\CHANEL_TryOn_08092022\Plugins\AzureKinectUnreal\Binaries\Win64\k4a.dll to D:\UPROJECTS\CHANEL_TryOn_08092022\Saved\StagedBuilds\WindowsNoEditor\Chanel_TryOn\Plugins\AzureKinectUnreal\Binaries\Win64\k4a.dll

**Comment by VegetableWithChicken** [comment]
> Hi,
> 
> I have similar issue here :
> 
> UATHelper: Packaging (Windows (64-bit)): ERROR: Failed to copy D:\UPROJECTS\CHANEL_TryOn_08092022\Plugins\AzureKinectUnreal\Binaries\Win64\k4a.dll to D:\UPROJECTS\CHANEL_TryOn_08092022\Saved\StagedBuilds\WindowsNoEditor\Chanel_TryOn\Plugins\AzureKinectUnreal\Binaries\Win64\k4a.dll UATHelper: Packaging (Windows (64-bit)): ERROR: Failed to copy D:\UPROJECTS\CHANEL_TryOn_08092022\Plugins\AzureKinectUnreal\Binaries\Win64\k4a.dll to D:\UPROJECTS\CHANEL_TryOn_08092022\Saved\StagedBuilds\WindowsNoEditor\Chanel_TryOn\Plugins\AzureKinectUnreal\Binaries\Win64\k4a.dll

change the file of AzureKinectUnreal.Build.cs ,
```
azure-kinect-unreal-master\Plugins\AzureKinectUnreal\Source\AzureKinectUnreal\AzureKinectUnreal.Build.cs
// Ensure that the DLLs are staged along with the executable
//RuntimeDependencies.Add("$(PluginDir)/Binaries/Win64/k4a.dll");
//RuntimeDependencies.Add("$(PluginDir)/Binaries/Win64/k4abt.dll");
//RuntimeDependencies.Add("$(PluginDir)/Binaries/Win64/depthengine_2_0.dll");
//RuntimeDependencies.Add("$(PluginDir) / Binaries / Win64 /opencv_world430.dll");

to exchange

RuntimeDependencies.Add(Path.Combine(AzureKinectBinaryPath, "k4a.dll"));
RuntimeDependencies.Add(Path.Combine(AzureKinectBinaryPath, "k4abt.dll"));
RuntimeDependencies.Add(Path.Combine(AzureKinectBinaryPath, "depthengine_2_0.dll"));
RuntimeDependencies.Add(Path.Combine(AzureKinectBinaryPath, "opencv_world430.dll"));

**Comment by VegetableWithChicken** [comment]
> Thank you for answer, I tried this solution as it was the solution of the original post.
> 
> But I have the following compile error: The name 'AzureKinectBinaryPath' does not exist in the current context


you can define it . it is no problem,and would work.

**Comment by jdamdami** [comment]
Ok managed to narrow down the issue.

The issue is coming from the fact that the dll files are not in the binaries plugin folder because I removed them.

I removed them because otherwise in PIE I have a kinect device not found error; This error can only be solved if I remove the dll for the kinect from the plugin binairies files. But when packaging, it says it can't copy the dll files as it does not find them where they should be.

Tried to bypass this issue by adding the dll to the binary folder and recompiling before trying to package which I don't believe is ideal but it's not working neither.

**Comment by VegetableWithChicken** [comment]
> Ok managed to narrow down the issue.
> 
> The issue is coming from the fact that the dll files are not in the binaries plugin folder because I removed them.
> 
> I removed them because otherwise in PIE I have a kinect device not found error; This error can only be solved if I remove the dll for the kinect from the plugin binairies files. But when packaging, it says it can't copy the dll files as it does not find them where they should be.
> 
> Tried to bypass this issue by adding the dll to the binary folder and recompiling before trying to package which I don't believe is ideal but it's not working neither.

you should copy the dll file in Project binaries but in Plugin Binaries

**Comment by jdamdami** [comment]
Yes that is what I did , I copied them in the plugin binaries folder.

I have these errors when I try to package : 

UATHelper: Packaging (Windows (64-bit)):   Script Stack (0 frames):
UATHelper: Packaging (Windows (64-bit)):   
UATHelper: Packaging (Windows (64-bit)):   Windows GetLastError: The operation completed successfully. (0)
UATHelper: Packaging (Windows (64-bit)):   begin: stack for UAT
UATHelper: Packaging (Windows (64-bit)):   === Critical error: ===
UATHelper: Packaging (Windows (64-bit)):   
UATHelper: Packaging (Windows (64-bit)):   Fatal error: [File:D:/Build/++UE4/Sync/Engine/Source/Runtime/Launch/Private/Windows/LaunchWindows.cpp] [Line: 95]
UATHelper: Packaging (Windows (64-bit)):   SECURE CRT: Invalid parameter detected.
UATHelper: Packaging (Windows (64-bit)):   Expression: Unknown Function: Unknown. File: Unknown Line: 0
UATHelper: Packaging (Windows (64-bit)):   
UATHelper: Packaging (Windows (64-bit)):   
UATHelper: Packaging (Windows (64-bit)):   
UATHelper: Packaging (Windows (64-bit)):   
UATHelper: Packaging (Windows (64-bit)):   end: stack for UAT
UATHelper: Packaging (Windows (64-bit)):   FPlatformMisc::RequestExit(1)
UATHelper: Packaging (Windows (64-bit)):   FPlatformMisc::RequestExitWithStatus(1, 3)
UATHelper: Packaging (Windows (64-bit)):   Engine exit requested (reason: Win RequestExit; note: exit was already requested)
PackagingResults: Error: The operation completed successfully. (0)
PackagingResults: Error: ===
PackagingResults: Error: [File:D:/Build/++UE4/Sync/Engine/Source/Runtime/Launch/Private/Windows/LaunchWindows.cpp] [Line: 95]
UATHelper: Packaging (Windows (64-bit)): Took 35.5414054s to run UE4Editor-Cmd.exe, ExitCode=3
UATHelper: Packaging (Windows (64-bit)): ERROR: Cook failed.
UATHelper: Packaging (Windows (64-bit)):        (see C:\Users\Admin\AppData\Roaming\Unreal Engine\AutomationTool\Logs\D+UENGINE+UE_4.27\Log.txt for full exception trace)
PackagingResults: Error: Cook failed.
UATHelper: Packaging (Windows (64-bit)): AutomationTool exiting with ExitCode=25 (Error_UnknownCookFailure)
UATHelper: Packaging (Windows (64-bit)): BUILD

**Comment by jdamdami** [comment]
Ok I managed to solve the issue.

What I did is that I simply commented out all the part concerning the copy of the dll in the packaged build, the build is now successful and the packaged application is working properly (I guess it's because it still manages to find the dependencies that it need even though there are not in the binaries folder)
